### ISSUE TITLE
Steam version can't open scripts in external editors

### ISSUE THREAD

**Issue by NoIE** [godotengine/godot#5614]
_Bugsquad edit:_ Repurposed the issue to handle the bug described in https://github.com/godotengine/godot/issues/5614#issuecomment-231552662, i.e. that the Steam version of Godot can't open external editors properly.

---

I hope that clicking the Script icon would open it in Third-party editor, like gedit.

So I can enter chinese comment to script.

**Comment by SuperUserNameMan** [comment]
the "External Editor" setting into "Editor settings" does not works on linux ?

**Comment by akien-mga** [comment]
Well it does actually :D My bad, @NoIE had asked about this on Steam and I thought we only supported code completion in external editors, but not opened them via the Script icon, so I asked them to propose it as a feature here. Should have checked before-hand :)

So @NoIE, just define the path to gedit in Editor Settings > External Editor, it works.

**Comment by NoIE** [comment]
Thank you!

it works, but only when I run ~/.local/share/Steam/steamapps/common/Godot Engine/godot.x11.opt.tools.64.

If I run godot with steam library, it still don't open gedit. I guess steam forbid game call local program.

**Comment by akien-mga** [comment]
Right, this is indeed a bug that we didn't know of yet. I tested on Mageia 6 x64 and I can confirm that loading an external editor won't work due to Steam's antique runtime. For example here I used `/usr/bin/kate` as external editor, and get:
```
/usr/bin/kate: /home/akien/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /lib64/libQt5Xml.so.5)
/usr/bin/kate: /home/akien/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /lib64/libQt5Widgets.so.5)
/usr/bin/kate: /home/akien/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /lib64/libQt5Gui.so.5)
/usr/bin/kate: /home/akien/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /lib64/libQt5DBus.so.5)
/usr/bin/kate: /home/akien/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu/libdbus-1.so.3: no version information available (required by /lib64/libQt5DBus.so.5)
[...]
```

**Comment by akien-mga** [comment]
To get it to work we'd need to workaround Steam's definition of `LD_LIBRARY_PATH`, but the current interface doesn't allow to set environment variable as far as I know.

Any idea @punto-?

**Comment by akien-mga** [comment]
I wonder if we could disable the steam runtime completely btw, as we don't really need it for Godot.

**Comment by akien-mga** [comment]
Right, running Godot with `LD_LIBRARY_PATH="" %command%` as launch option in the properties fixes the issue. @punto- Is it possible to deactivate the Steam runtime in the deployment options?

**Comment by Calinou** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
Sadly fixing this is outside our control, the Steam runtime has to be kept up to date with recent Linux distros.

This specific issue might be fixed by now though, as the Steam runtime *was* updated several times.
